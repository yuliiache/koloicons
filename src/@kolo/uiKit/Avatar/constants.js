export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
