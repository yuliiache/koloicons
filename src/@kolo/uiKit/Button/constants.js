export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
