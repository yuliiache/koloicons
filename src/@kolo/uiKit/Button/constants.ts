export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
