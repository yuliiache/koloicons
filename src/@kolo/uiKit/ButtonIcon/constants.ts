export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
