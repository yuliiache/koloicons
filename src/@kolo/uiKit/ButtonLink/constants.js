export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

