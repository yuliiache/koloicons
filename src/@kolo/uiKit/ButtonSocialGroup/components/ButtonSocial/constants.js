export const BUTTON_SOCIAL_TEST_ID = 'buttonSocialComponent';

export const BUTTON_SOCIAL_TEXT_TEST_ID = `${BUTTON_SOCIAL_TEST_ID}Text`;
