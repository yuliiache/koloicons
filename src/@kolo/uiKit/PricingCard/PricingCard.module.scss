@import "@kolo/design";

.card {
  width: calc(33.33333% - 32px);
  margin: 0 16px 16px;
  padding: 24px 24px 16px;
  background-color: $WhiteBG;
  box-shadow: $PricingCardShadow;
  font-size: 14px;
}



