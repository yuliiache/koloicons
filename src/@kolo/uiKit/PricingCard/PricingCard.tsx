import {FC} from 'react';

import Button from '../Button/Button';
import {ButtonSize} from '../Button/constants';
import {LabelTheme} from '../Label/constants';
import Features from './Features/Features';
import styles from './PricingCard.module.scss';
import SubscriptionDetails from './SubscriptionDetails/SubscriptionDetails';
import SubscriptionPlan from './SubscriptionPlan/SubscriptionPlan';

interface PricingCardProps {
  name: string;
  price: string;
  theme: LabelTheme;
  periodInfo: string;
  planFeatures: string[];
  allowedOptions: string[];
  disallowedOptions: string[];
}

const PricingCard: FC<PricingCardProps> = ({
  name,
  price,
  theme,
  periodInfo,
  planFeatures,
  allowedOptions,
  disallowedOptions,
}) => {
  const purchaseHandle = () => {
    console.log(`The Purchase button was clicked with the next subscription plan: ${planFeatures}`);
  };

  return (
    <div className={styles.card}>
      <SubscriptionPlan
        name={name}
        price={price}
        theme={theme}
        periodInfo={periodInfo}
      />
      <Features planFeatures={planFeatures} />
      <Button
        text="Purchase"
        size={ButtonSize.FULL_WIDTH}
        onClick={purchaseHandle}
      />
      <SubscriptionDetails
        formats={['PNG', 'SVG']}
        attribution="No attribution required"
        allowedOptions={allowedOptions}
        disallowedOptions={disallowedOptions}
      />
    </div>
  );
};

export default PricingCard;
