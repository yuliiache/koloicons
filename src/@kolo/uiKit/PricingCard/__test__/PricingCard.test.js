import {LabelTheme} from '@kolo/uiKit/Label/constants';
import {cleanup, render} from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';

import PricingCard from '../PricingCard';

describe('PricingCard Component', () => {
  afterEach(cleanup);

  const commonProps = {
    name: 'Personal',
    price: '$2.50',
    theme: LabelTheme.SUCCESS,
    periodInfo: 'per month, $30 billed yearly',
    planFeatures: ['Feature 1', 'Feature 2', 'Feature 3'],
    allowedOptions: ['Option 1', 'Option 2', 'Option 3'],
    disallowedOptions: ['Option 4', 'Option 5', 'Option 6'],
  };

  it('renders snapshot PricingCard component', () => {
    const {asFragment} = render(<PricingCard {...commonProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders PricingCard with correct content', () => {
    const {getByText} = render(<PricingCard {...commonProps} />);

    expect(getByText(commonProps.name)).toBeInTheDocument();
    expect(getByText(commonProps.price)).toBeInTheDocument();
    expect(getByText(commonProps.periodInfo)).toBeInTheDocument();

    commonProps.planFeatures.forEach((feature) => {
      expect(getByText(feature)).toBeInTheDocument();
    });

    commonProps.allowedOptions.forEach((option) => {
      expect(getByText(option)).toBeInTheDocument();
    });

    commonProps.disallowedOptions.forEach((option) => {
      expect(getByText(option)).toBeInTheDocument();
    });
  });
});
