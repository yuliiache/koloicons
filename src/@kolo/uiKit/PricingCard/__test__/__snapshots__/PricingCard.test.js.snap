// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingCard Component renders snapshot PricingCard component 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <span
      class="label success large"
      data-testid="labelComponent"
    >
      Personal
    </span>
    <div
      class="price"
    >
      <span
        class="priceVal"
      >
        $2.50
      </span>
      <span
        class="priceDesc"
      >
        per month, $30 billed yearly
      </span>
    </div>
    <ul
      class="planFeatures"
    >
      <li>
        <strong>
          Feature 1
        </strong>
      </li>
      <li>
        <strong>
          Feature 2
        </strong>
      </li>
      <li>
        <strong>
          Feature 3
        </strong>
      </li>
    </ul>
    <button
      class="button button-full-width button-primary"
      data-testid="buttonComponent"
      type="button"
    >
      <span
        data-testid="buttonComponentText"
      >
        Purchase
      </span>
    </button>
    <div
      class="advantages"
    >
      <p
        class="advantagesItem"
      >
        <span
          class="itemTitle"
        >
          Formats:
        </span>
        <span
          class="itemVal"
        >
          PNG, SVG
        </span>
      </p>
      <p
        class="advantagesItem"
      >
        <span
          class="itemTitle"
        >
          Attribution:
        </span>
        <span
          class="itemVal"
        >
          No attribution required
        </span>
      </p>
    </div>
    <dl
      class="advantages"
    >
      <dt
        class="optionTitle"
      >
        You can use icons for:
      </dt>
      <dd
        class="optionVal"
      >
        <span>
           - 
        </span>
        Option 1
      </dd>
      <dd
        class="optionVal"
      >
        <span>
           - 
        </span>
        Option 2
      </dd>
      <dd
        class="optionVal"
      >
        <span>
           - 
        </span>
        Option 3
      </dd>
      <dt
        class="optionTitle"
      >
        You can not use icons for:
      </dt>
      <dd
        class="optionVal"
      >
        <span>
           - 
        </span>
        Option 4
      </dd>
      <dd
        class="optionVal"
      >
        <span>
           - 
        </span>
        Option 5
      </dd>
      <dd
        class="optionVal"
      >
        <span>
           - 
        </span>
        Option 6
      </dd>
    </dl>
  </div>
</DocumentFragment>
`;
